### PR TITLE
test(common): improve test coverage for shared utils

### PR DESCRIPTION
**Pull request by BrahimAbdelli**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?
Add tests for edge cases in shared utility functions
Adding type safety
Update normalizePath, isEmpty, stripEndSlash

Issue Number: N/A

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

### CODE REVIEW

File: packages/common/utils/shared.utils.ts [hunk 1]
```diff
@@ -37,15 +37,25 @@ export const normalizePath = (path?: string): string =>
...
+
+export const isConstructor = (val: unknown): boolean => val === 'constructor';
+
+export const isNil = (val: unknown): val is null | undefined =>
   isUndefined(val) || val === null;
-export const isEmpty = (array: any): boolean => !(array && array.length > 0);
-export const isSymbol = (val: any): val is symbol => typeof val === 'symbol';
+
+export const isEmpty = (array: unknown): boolean =>
+  !(Array.isArray(array) && array.length > 0);
```
Comment (micalevisk) [inline packages/common/utils/shared.utils.ts:58]: to me, `isEmpty` seems that it could be used on non-array values...

peharps we could rename this utility to `isEmptyArray` and drop the following:

https://github.com/nestjs/nest/blob/e1b91d02a601c03cb8d0438b32badfaae5403447/packages/common/pipes/file/parse-file.pipe.ts#L63

**Review by micalevisk — APPROVED** [review]

**Comment by BrahimAbdelli** [comment]
Hi @kamilmysliwiec , is there anything else needed to get this merged? Happy to help if further changes are required !

**Comment by kamilmysliwiec** [comment]
Hey @BrahimAbdelli, everything looks great! I flagged this PR as "blocked" as there're a few changes that _may_ introduce _minor breaking changes_ in case someone relied on these utils in community packages

**Comment by kamilmysliwiec** [comment]
Could you rebase to v12.0.0 branch?

**Comment by BrahimAbdelli** [comment]
@kamilmysliwiec Rebased onto v12.0.0 as requested.

**Comment by kamilmysliwiec** [comment]
looks like the PR is now in a broken state

**Comment by kamilmysliwiec** [comment]
tests fail now

**Comment by BrahimAbdelli** [comment]
@kamilmysliwiec  should be good now
